tests/functional/settings_secondary_emails.js

```diff
@@ -60,6 +60,49 @@ define([
       return this.remote.then(clearBrowserState());
     },
 
+    'gated in unverified session open verification same tab': function () {
+      return this.remote
+        // when an account is created, the original session is verified
+        // re-login to destroy original session and created an unverified one
+        .then(createUser(email, PASSWORD, { preVerified: true }))
+        .then(openPage(SIGNIN_URL, selectors.SIGNIN.HEADER))
+        .then(fillOutSignIn(email, PASSWORD))
+        .then(testElementExists(selectors.EMAIL.UNLOCK_BUTTON))
+
+        // unlock panel
+        .then(click(selectors.EMAIL.UNLOCK_BUTTON))
+        .then(testElementExists(selectors.EMAIL.UNLOCK_REFRESH_BUTTON))
+
+        // send and open verification in same tab
+        .then(click(selectors.EMAIL.UNLOCK_SEND_BUTTON))
+        .then(openVerificationLinkInSameTab(email, 0))
+
+        // panel becomes verified and opens add secondary panel
+        .then(testElementExists(selectors.EMAIL.INPUT));
+    },
+
+    'gated in unverified session open verification new window': function () {
+      return this.remote
+        // when an account is created, the original session is verified
+        // re-login to destroy original session and created an unverified one
+        .then(createUser(email, PASSWORD, { preVerified: true }))
+        .then(openPage(SIGNIN_URL, selectors.SIGNIN.HEADER))
+        .then(fillOutSignIn(email, PASSWORD))
+        .then(testElementExists(selectors.EMAIL.UNLOCK_BUTTON))
+
+        // unlock panel
+        .then(click(selectors.EMAIL.UNLOCK_BUTTON))
+        .then(testElementExists(selectors.EMAIL.UNLOCK_REFRESH_BUTTON))
+
+        // send and open verification in same tab
+        .then(click(selectors.EMAIL.UNLOCK_SEND_BUTTON))
+        .then(openVerificationLinkInDifferentBrowser(email, 0))
```

Comment:
One test not covered here is what the verification page looks like from the new browser. I see "Sign-in confirmed". Is that expected?

<img width="458" alt="screen shot 2017-11-08 at 14 46 35" src="https://user-images.githubusercontent.com/848085/32555122-d6167142-c493-11e7-81b1-16c238bbaee4.png">

I thought it should use the new [PRIMARY_EMAIL_VERIFIED mesage](https://github.com/mozilla/fxa-content-server/pull/5626/files#diff-8d6c56ddab1199e1b057cdfc6d77a9c1R43).

Reply:
Strange, it should. Are you using the `master` auth server?

Reply:
Yup.
<img width="578" alt="screen shot 2017-11-08 at 15 03 39" src="https://user-images.githubusercontent.com/848085/32555999-1d4f14e0-c496-11e7-8ac3-e8fb3b41d455.png">